master/internal/experiment_job_service.go

```diff
@@ -79,3 +79,10 @@ func (e *internalExperiment) SetResourcePool(resourcePool string) error {
 
 	return e.setRP(resourcePool)
 }
+
+// GetResourcePool gets the experiment's resource pool.
+func (e *internalExperiment) ResourcePool() string {
+	e.mu.Lock()
+	defer e.mu.Lock()
+	return e.getRP()
```

Comment:
i think it is fine to not have e.getRP() and just call  e.activeConfig.Resources().ResourcePool() righ there